File: src/test/java/no/nav/syfo/config/mq/mock/JmsTemplateMock.java
package no.nav.syfo.config.mq.mock;

import org.slf4j.*;
import org.springframework.jms.*;
import org.springframework.jms.core.*;

import javax.jms.*;

public class JmsTemplateMock extends JmsTemplate {
    private static final Logger log = LoggerFactory.getLogger(JmsTemplateMock.class);

    private String name;

    public JmsTemplateMock(String name) {
        super(new ConnectionFactoryMock());
        this.name = name;
    }

    @Override
    public void send(MessageCreator messageCreator) throws JmsException {
        log.info("Sender melding til {}", name);
        try {
            TextMessage message = (TextMessage) messageCreator.createMessage(new SessionMock());
            log.info("Call id: {}", message.getStringProperty("callId"));
            log.info("Text:\n{}", message.getText());
        } catch (JMSException e) {
            throw new UncategorizedJmsException(e);
        }
    }
}

File: src/test/kotlin/no/nav/syfo/testhelper/mock/mq/JmsTemplateMock.kt
package no.nav.syfo.testhelper.mock.mq

import org.slf4j.LoggerFactory
import org.springframework.jms.JmsException
import org.springframework.jms.UncategorizedJmsException
import org.springframework.jms.core.JmsTemplate
import org.springframework.jms.core.MessageCreator
import javax.jms.JMSException
import javax.jms.TextMessage

class JmsTemplateMock(
    private val name: String
) : JmsTemplate(ConnectionFactoryMock()) {
    @Throws(JmsException::class)
    override fun send(messageCreator: MessageCreator) {
        log.info("Sender melding til {}", name)
        try {
            val message = messageCreator.createMessage(SessionMock()) as TextMessage
            log.info("Call id: {}", message.getStringProperty("callId"))
            log.info("Text:\n{}", message.text)
        } catch (e: JMSException) {
            throw UncategorizedJmsException(e)
        }
    }

    companion object {
        private val log = LoggerFactory.getLogger(JmsTemplateMock::class.java)
    }
}
